nixos/tests/radicle.nix

```diff
@@ -0,0 +1,187 @@
+# This test runs a Radicle seed on one host, and verifies
+# that two peer hosts can exchange a patch over that Radicle seed.
+
+import ./make-test-python.nix ({ pkgs, ... }:
+
+let
+  seedNID = "z6MkqZDt6RJ2tUsq4j2F3RSB6Xg8Pcw2KgPXUqxd1YwFRK6D";
+
+  commonHostConfig = { config, pkgs, ... }: {
+    networking.domain = "localdomain";
+    environment.systemPackages = [
+      pkgs.jq
+    ];
+    environment.etc."gitconfig".text = ''
+      [init]
+        defaultBranch = main
+      [user]
+        email = root@${config.networking.hostName}.localdomain
+        name = ${config.networking.hostName}.localdomain
+    '';
+  };
+
+  radicleConfig = { nodes, ... }: alias:
+    pkgs.writeText "config.json" (builtins.toJSON {
+      preferredSeeds = [
+        "${seedNID}@seed.localdomain:${toString nodes.seed.services.radicle.node.listenPort}"
+      ];
+      node = {
+        inherit alias;
+        relay = "never";
+        policy = "block";
+        scope = "all";
+      };
+    });
+in
+
+{
+  name = "radicle";
+
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [
+      julm
+      lorenzleutgeb
+    ];
+  };
+
+  nodes = {
+    seed = { pkgs, config, ... }: {
+      imports = [ commonHostConfig ];
+
+      services.radicle = {
+        enable = true;
+        privateKeyFile = pkgs.writeText "id_ed25519" ''
+          -----BEGIN OPENSSH PRIVATE KEY-----
+          b3BlbnNzaC1rZXktdjEAAAAABG5vbmUAAAAEbm9uZQAAAAAAAAABAAAAMwAAAAtzc2gtZW
+          QyNTUxOQAAACCk95gwwCfu2ezgoPOHsATixXUqvm7k2mfNx6t3lBf71gAAAJDAzemhwM3p
+          oQAAAAtzc2gtZWQyNTUxOQAAACCk95gwwCfu2ezgoPOHsATixXUqvm7k2mfNx6t3lBf71g
+          AAAECMhq3OzhskoKMnQ6nMExCcRIxchCHX75shzdy0qfJcUaT3mDDAJ+7Z7OCg84ewBOLF
+          dSq+buTaZ83Hq3eUF/vWAAAAB3JhZGljbGUBAgMEBQY=
+          -----END OPENSSH PRIVATE KEY-----
```

Comment:
Instead of checking in a keypair to Nixpkgs, maybe reuse one?

 - https://github.com/NixOS/nixpkgs/blob/f478e1cb40ff17ef02ab22f45e0512868b0c86c6/nixos/tests/ssh-keys.nix
 - https://github.com/NixOS/nixpkgs/tree/f478e1cb40ff17ef02ab22f45e0512868b0c86c6/nixos/modules/profiles/keys

Reply:
you're right, I hesitated to use `ssh-keys.nix` for no good reason.